packages/updating-element/src/updating-element.ts

```diff
@@ -544,6 +556,17 @@ export abstract class UpdatingElement extends HTMLElement {
     this.requestUpdate();
   }
 
+  addCallbacks(callbacks: LifecycleCallbacks) {
+    if (this._callbacks === undefined) {
+      this._callbacks = new Set();
+    }
+    this._callbacks.add(callbacks);
+  }
+
+  removeCallbacks(callbacks: LifecycleCallbacks) {
+    this._callbacks!.delete(callbacks);
```

Comment:
This should use nullish coalescing in case the collection hasn't been created yet:

```suggestion
    this._callbacks?.delete(callbacks);
```

If this were an array, it'd be a bit longer, but not too much:

```suggestion
    this._callbacks = this._callbacks?.filter((c) => c !== callbacks);
```

Removal should be _exceedingly_ rare, so we should optimize for rendering then adding then removing. The code size is probably the only significant factor on removal. Iteration cost _might_ matter - I bet the common case is just one or two controllers.

The nice thing about the `addCallbacks()` approach is that the storage is abstracted over, so we can change this over time.

Reply:
Done. Now that the Set is an implementation detail, it should be easy to revisit this.

Reply:
Changed to array.